### DISCUSSION TITLE
future times on xAxis

### DISCUSSION THREAD

**Discussion by mawsyh** [klinecharts/KLineChart#415] [Ideas]
I am thinking of adding future timestamps (the ones in users viewport) to the xAxis.
Any tips or ideas ?

**Comment by liihuu** [comment]
This is difficult because the chart is not currently associated with the trading time period.

**Comment by mawsyh** [comment]
![image](https://user-images.githubusercontent.com/90469267/234795225-3f8e9af1-9e26-4a8c-b011-4a36c8c24c50.png)

I implemented this to my code should I ask for a PR so you can check it out ?

**Reply by liihuu** [reply]
Is it filled based on the time period? If so, how to determine the current period?

**Reply by mawsyh** [reply]
It is, the point was that we didn't have to know the time span, all I needed was the past xAxis times and the difference in coord and timestamp values (because it stays the same). by calculating these values we are able to add any amount of future times on xScale on any period. (Sorry for the delay on answer, let me know If this explains your concerns)

**Reply by liihuu** [reply]
It can be calculated, but it seems to only fit the trading time period of 7 * 24.

**Reply by mawsyh** [reply]
can you check this PR out please : https://github.com/liihuu/KLineChart/pull/421
it works fine for me on every timespan
